src/org.openjdk.nashorn/share/classes/org/openjdk/nashorn/internal/lookup/MethodHandleFactory.java

```diff
@@ -369,13 +369,11 @@ protected static String describe(final Object... data) {
         }
 
         public MethodHandle debug(final MethodHandle master, final String str, final Object... args) {
-            if (log.isEnabled()) {
-                if (PRINT_STACKTRACE) {
-                    stacktrace(log);
-                }
-                return addDebugPrintout(log, Level.INFO, master, Integer.MAX_VALUE, false, str + ' ' + describe(args));
+            // The static `log` field can be `null` if the `MethodHandlerFactory` class has not been initialized yet.
+            if (PRINT_STACKTRACE) {
```

Comment:
`stacktrace` itself checks `PRINT_STACKTRACE` so this `if` could also be removed.

Reply:
Fixed in https://github.com/openjdk/nashorn/pull/19/commits/ee8edb3f800c9f09753a25d53e6b05633ca5e62d